### PR TITLE
Feature/multi line docstrings

### LINKED ISSUE #83: Add check for multiple-line Python docstring comments
Although it is possible to have multiple-line comments in Python in the form of docstrings that are part of, say, a function's description, the GatorGrader tool does not have a check for these. We should add this feature to the tool so that it is possible to ensure that a Python program has an adequate number of docstrings. We would need to either add a regular expression for docstrings or perhaps use the Python parser and then extract. At this point, we would need to ensure that the number of docstrings matches the specification for the check. The implementation of this feature will also require the support of Python as a language for the multiple-line check.

### PR DESCRIPTION
**Pull request by mendezjw**
# Description of Pull Request

Fixes #83  

This PR contains the following:

- Adds check for python multiple-line docstrings comments.
- Adds test cases for checking this feature.

## Type of Change

Please describe the pull request as one of the following: 

- [ ] Bug fix
- [ ] Breaking change
- [x] New feature
- [ ] Documentation update
- [X] Other (Refactoring and Testing)
## Tags

@ilikerustoo @JattMones @wattob @thomad74

### CODE REVIEW

**Comment by wattob** [comment]
Our feature still fails to pass Travis. It also appears we have issues with a conflicting `Pipfile.lock`. I am not sure what these means. Does anyone have a suggestion?

**Comment by thomad74** [comment]
@Michionlion Do you have any idea how to solve this issue?

**Comment by Michionlion** [comment]
@thomad74 @mendezjw So sorry I haven't been able to get to this, but I just fixed the issues relating to not being able to run `pipenv run cover`. Now you can actually get results from `pytest`. You should use this complete the following:

* Fix tests to actually test ***python*** comments -- currently they are still java single line, which must be changed. Almost all of your parameters for your parametrized test cases will need to be changed.
* Add a test to `tests/test_invoke.py` to test your invoke method (similar to existing tests for other checks)
* Add a test to `tests/test_orchestrate.py` to test the command line argument handling (similar to the existing check for the multiline java comment command)
* Add parameters to the empty parametrize that was added; these should be similar to the existing ones, except adapted for python.

These tasks should be completed by Wednesday morning -- @gkapfham and I will be reviewing PRs and merging at that time, and it would be great to merge this feature. However, that is not possible without better tests.

File: gator/comments.py [hunk 1]
```diff
@@ -34,3 +35,10 @@ def count_multiline_java_comment(contents):
     pattern = re.compile(MULTILINECOMMENT_RE, re.MULTILINE)
     matches = pattern.findall(contents)
     return len(matches)
+
+
+def count_multiline_python_comment(contents):
+    """Counts the number of multiline Python comments in the code"""
+    pattern = re.compile(MULTILINECOMMENT_RE_PYTHON, re.DOTALL)
```
Comment (Michionlion) [inline gator/comments.py]: This needs `re.MULTILINE`, and `re.DOTALL` should be removed -- that enables `.` substituting for new lines, which you do not want with your current regex; you could change the regex to enable that however, if you wanted.

File: gator/comments.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 \"|/(?:[^/"\\]|\\.)|/\"(?:[^\"\\]|\\.)*\"|\\.)*//(.*)$"""
 SINGLELINECOMMENT_RE_PYTHON = r"""^(?:[^"#\\]|\"(?:[^\"\\]|\\.)*\"|
 /(?:[^#"\\]|\\.)|/\"(?:[^\"\\]|\\.)*\"|\\.)*#(.*)$"""
+MULTILINECOMMENT_RE_PYTHON = r'^[\s]*"""(.*?\n|.*?)*?"""'
```
Comment (Michionlion) [inline gator/comments.py]: I forgot this earlier, but you should probably have a `$` to end the regex -- that stands for "end of line". You may want to allow trailing whitespace as well.

File: tests/test_comments.py [hunk 3]
```diff
@@ -81,7 +109,23 @@ def test_file_contains_multiline_comment_not_greater(tmpdir):
...
+        comments.count_multiline_java_comment,
+    )
+    assert greater_than_count is False
+
+
+def test_file_contains_multiline_python_comment_not_greater(tmpdir):
+    """Checks that the file is not above the check level"""
+    hello_file = tmpdir.mkdir("subdirectory").join("Hello.py")
+    string = "Hello \n World"
+    hello_file.write('"""{}"""'.format(string))
```
Comment (Michionlion) [inline tests/test_comments.py]: This should be zero comments, not one.

File: tests/test_comments.py [hunk 4]
```diff
@@ -268,3 +374,31 @@ def test_multiline_comments_mixed(code_string, expected_count):
 def test_singleline_comments_mixed(code_string, expected_count):
     """Checks that it can find singleline comments in mixtures"""
     assert comments.count_singleline_java_comment(code_string) == expected_count
+
+@pytest.mark.parametrize(
+    "code_string,expected_count",
+    [
+        ("hello world", 0)
```
Comment (Michionlion) [inline tests/test_comments.py]: Needs more test cases.

**Comment by mendezjw** [comment]
**Status Update:**

I think I have implemented the **_invoke_** and the **_orchestrate_** tests correctly, however I am unable to currently tell due to some of the tests in **_comments_** running in infinite loops as of the most updated version of that file. I am unsure if this is due to my current regex or the current quality of those test cases and may require a roll back if I determine the issues have gotten too complex to figure out. I don't have anymore time to look at these issues tonight, so I'll try to get to them some point in the morning or when I have a chance tomorrow.

**Comment by Michionlion** [comment]
@mendezjw I'll take a look -- the merge date has been pushed back due to scheduling things, so you'll have a longer time.

**Comment by Michionlion** [comment]
@mendezjw, nevermind, I got carried away and fixed everything. Most things were just minor changes, but the regex was not really right, and the biggest change was fixing that. I had to fix most of the test cases to work with it as well.

One thing is that we are now expecting only whitespace before and after the docstring (on the same line as the beginning/ending).

This should be good to go, but I would appreciate a few reviews from others to ensure everything is good!

File: tests/test_comments.py (hunk 3 above)
Comment (Michionlion) [reply tests/test_comments.py]: This was actually a bug (the java method was being called), I fixed it.

**Comment by quigley-c** [comment]
> @mendezjw, nevermind, I got carried away and fixed everything.

That's our @Michionlion xD

**Pull request merged**